awx/main/tasks.py

```diff
@@ -494,29 +495,22 @@ def execution_node_health_check(node):
 
 def inspect_execution_nodes(instance_list):
     with advisory_lock('inspect_execution_nodes_lock', wait=False):
-        node_lookup = {}
-        for inst in instance_list:
-            if inst.node_type == 'execution':
-                node_lookup[inst.hostname] = inst
+        # {inst.hostname: inst for inst in instance_list}
+        node_lookup = Instance.objects.in_bulk(field_name='hostname')
```

Comment:
What I don't like about this is that it would (presumably) incur an additional query. Normally, 1 query isn't something I should raise a stink about. But this code path is the heartbeat, which has very poor scaling for clusters with lots of nodes and just 1 database node. That's why this code was kind of awkward to begin with, this is a special case where performance is a priority.

Reply:
You are right.  Rolled back to a dict comprehension.